torchtune/datasets/_finetune.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, Mapping
+
+import numpy as np
+
+from datasets import load_dataset
+from torch.utils.data import Dataset
+from torchtune.data import CROSS_ENTROPY_IGNORE_IDX
+from torchtune.modules.transforms import Transform
+
+
+class FinetuneDataset(Dataset):
```

Comment:
I don't like the idea of differentiating between a "finetune" dataset and any other kind of dataset - doesn't feel like the right name. 

Still working out what would be the best convention here.

Reply:
Yeah I couldn't come to a good name. I did "finetune" because preference = RLHF, text completion = CPT, and everything else falls under finetune. Maybe could keep it as ChatDataset? MessageDataset?

Reply:
Related to the above point, I do find that instruct/chat/text-completion dataset notion is easy to understand. Would it make sense to still provide those as higher level APIs that build above this new implementation? That way we kind of meet users wherever they are. 1) with specific implemnentaitno like alpaca. 2) with instruct/chat/text-completion and 3) for advanced users, just the base notion of a TorchTuneDataset